test/parallel/test-tls-env-extra-ca-with-ca.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+const common = require('../common');
+
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+
+const assert = require('assert');
+const tls = require('tls');
+const fixtures = require('../common/fixtures');
+
+const { fork } = require('child_process');
+
+if (process.env.CHILD) {
+  const copts = {
+    port: process.env.PORT,
+    checkServerIdentity: common.mustCall()
```

Comment:
Nitpicking: Given that all 3 of these tests are basically identical except for a single option here, these might be better as a single file with 3x fork tests below that just pass an option to change the one setting here. You'd probably want to change a few `mustCall()` to `mustCall(n)` accordingly.

Just to be thorough, it'd be nice to also include a failing test here too (not passing NODE_EXTRA_CA_CERTS) as in the previous test that's now deleted.

What do you think?

Reply:
I opted for the 'copy & paste' approach to keep the logic simpler, even if it's less elegant. The forking already seemed somewhat fragile - unit tests that were easier to read and understand seemed advantageous here. Particularly given the confusing 1/2/3 filenames, as noted earlier!

A reusable fixture for the forking, potentially part of `fixtures/tls-connect.js` would be a better long term solution, but I opted to keep my changes as minimal as possible as a relatively new contributor.

Good idea about the negative unit test - I'll see what I can do!

Reply:
I went ahead and combined the unit tests into a single file after all.

Didn't add a negative test case - the one that was deleted was specific to loading certs at startup, which is no longer the case. Any other negative test with NODE_EXTRA_CA_CERTS missing would end up being a regular TLS verification failure, which should already be handled by the general TLS unit test suite.

Let me know what you think though!